### PR TITLE
Port tests from Promise.all to Promise.allSettled

### PR DESCRIPTION
**Pull request by leobalter**
Most of the spec tests are pretty similar, so I've been porting the tests over. This still requires modding things here and there.

### CODE REVIEW

**Comment by leobalter** [comment]
I only need to finish reviewing my own tests and it's good for a review

**Comment by leobalter** [comment]
This is now all set! 

[v8 with the --harmony flag](https://circleci.com/gh/tc39/test262/544?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) is pretty close to have full compliance, nice job! cc @gsathya @mathiasbynens

**Comment by mathiasbynens** [comment]
@gsathya Could you take a look at the failing tests please?

**Comment by mathiasbynens** [comment]
cc @jasonwilliams @rpamely

**Comment by mathiasbynens** [comment]
Ref. https://github.com/tc39/proposal-promise-allSettled/issues/11.
Ref. https://github.com/tc39/test262/issues/2112.

**Comment by leobalter** [comment]
@mathiasbynens @gsathya I think we should limit the resolve lookup to only happen if the iterable has at least one item to iterate with. This requires some fix in the spec text. WDYT?

Current status (shown in the tests) will always get the ` resolve` once even if iterable is empty.

**Comment by leobalter** [comment]
The failing test checking the built-in function name is also happening in the tests for Promise.all

**Comment by gsathya** [comment]
> @gsathya Could you take a look at the failing tests please?

One is the unimplemented optimization (https://github.com/tc39/proposal-promise-allSettled/pull/40) and the other is a known web compat issue with function name.

> The failing test checking the built-in function name is also happening in the tests for Promise.all

Yeah, that's a known inconsistency. More background here: 
https://bugs.chromium.org/p/v8/issues/detail?id=4709
https://github.com/tc39/ecma262/issues/1049

File: test/built-ins/Promise/allSettled/call-resolve-element-after-return.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  Cannot change result value of resolved Promise.allSettled element after Promise.allSettled() returned.
+info: |
+  Promise.allSettled Resolve Element Functions
+
+  1. Let F be the active function object.
+  2. Let alreadyCalled be F.[[AlreadyCalled]].
+  3. If alreadyCalled.[[Value]] is true, return undefined.
+  4. Set alreadyCalled.[[Value]] to true.
+  ...
+includes: [promiseHelper.js]
```
[the remark this answers is not in the document]
Comment (leobalter) [reply test/built-ins/Promise/allSettled/call-resolve-element-after-return.js]: Not anymore 2b4d328

**Pull request merged**